apps/shelter/src/pages/signup/index.tsx

```diff
@@ -14,69 +16,179 @@ import {
   Switch,
   VStack,
 } from '@chakra-ui/react';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Controller, useForm } from 'react-hook-form';
 import AnimalfriendsLogo from 'shared/assets/image-anifriends-logo.png';
 import IoEyeOff from 'shared/assets/IoEyeOff';
 import IoEyeSharp from 'shared/assets/IoEyeSharp';
 import useToggle from 'shared/hooks/useToggle';
+import * as z from 'zod';
+
+type Schema = z.infer<typeof schema>;
+
+const schema = z
+  .object({
+    email: z
+      .string()
+      .min(1, '이메일은 필수 정보입니다')
+      .email('유효하지 않은 이메일입니다'),
+    password: z
+      .string()
+      .regex(
+        /^(?=.*[!@#$%^&*()\-_=+[\]\\|{};:'",<.>/?]+)(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{8,}$/,
+        '비밀번호는 필수 정보입니다(8자 이상)',
+      ),
+    passwordConfirm: z.string().min(1, '비밀번호 확인 정보는 필수입니다'),
+    name: z.string().min(1, '보호소 이름 정보는 필수입니다'),
+    address: z.string().min(1, '보호소 주소 정보는 필수입니다'),
+    addressDetail: z.string().min(1, '보호소 상세주소 정보는 필수입니다'),
+    isOpendedAddress: z.boolean(),
+    phoneNumber: z
+      .string()
+      .min(1, '보호소 전화번호 정보는 필수입니다')
+      .refine(
+        (val) => !Number.isNaN(Number(val)),
+        '전화번호 형식은 숫자입니다',
+      ),
+    sparePhoneNumber: z
+      .string()
+      .refine(
+        (val) => !Number.isNaN(Number(val)) || val === '',
+        '전화번호 형식은 숫자입니다',
+      ),
+  })
+  .refine(({ password, passwordConfirm }) => password === passwordConfirm, {
+    message: '비밀번호가 일치하지 않습니다',
+    path: ['passwordConfirm'],
+  });
 
 export default function SignupPage() {
   const [isPasswordShow, togglePasswordShow] = useToggle();
   const [isPasswordConfirmShow, togglePasswordConfirmShow] = useToggle();
-  const [isShelterAddressOpen, toggleShelterAddressOpen] = useToggle();
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    control,
+  } = useForm<Schema>({
+    defaultValues: {
+      isOpendedAddress: false,
+    },
+    resolver: zodResolver(schema),
+  });
+
+  const onSubmit = (data: Schema) => {
+    console.log(data);
+  };
 
   return (
     <Box px={4} pb="104px">
       <Center w="100%" py={10}>
         <Image boxSize={160} borderRadius={100} src={AnimalfriendsLogo} />
       </Center>
-      <form>
-        <FormControl mb={2} isRequired>
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <FormControl mb={2} isRequired isInvalid={errors.email ? true : false}>
```

Comment:
```suggestion
        <FormControl mb={2} isRequired isInvalid={Boolean(errors.email)}>
```

Boolean() 메서드를 활용한다면 코드의 의도가 좀 더 명시적으로 표현될 수 있을 것 같습니다~!